src/Reflex/Dom/Widget/Input.hs

```diff
@@ -212,6 +214,110 @@ dropdown k0 options (DropdownConfig setK attrs) = do
   dValue <- combineDyn readKey options =<< holdDyn (Just k0) (leftmost [eChange, fmap Just setK])
   return $ Dropdown dValue (attachDynWith readKey options eChange)
 
+
+data ButtonGroup t a
+   = ButtonGroup { _buttonGroup_value    :: Dynamic t (Maybe a)
+                 , _buttonGroup_select   :: Event  t (Maybe a)
+                 , _buttonGroup_element  :: El t
+                 , _buttonGroup_children :: Dynamic t (Map.Map Int (El t))
+                 }
+
+data ButtonGroupConfig t a
+   = ButtonGroupConfig { _buttonGroupConfig_parentType   :: String
+                       , _buttonGroupConfig_parentTag    :: String
+                       , _buttonGroupConfig_initialValue :: Maybe a
+                       , _buttonGroupConfig_setValue     :: Event t (Maybe a)
+                       , _buttonGroupConfig_attributes   :: Dynamic t (Map String String)
+                       }
+
+instance Reflex t => Default (ButtonGroupConfig t a) where
+  def = ButtonGroupConfig { _buttonGroupConfig_parentType   = ""
+                          , _buttonGroupConfig_parentTag    = "div"
+                          , _buttonGroupConfig_initialValue = Nothing
+                          , _buttonGroupConfig_setValue     = never
+                          , _buttonGroupConfig_attributes   = constDyn mempty
+                          }
+
+buttonGroup :: (MonadWidget t m, Eq a, Show a)
+            => (Maybe Int -> Dynamic t a -> Dynamic t Bool -> m (Event t (), El t))
+            -> Dynamic t (Map.Map Int a)
+            -> ButtonGroupConfig t a
+            -> m (ButtonGroup t a)
+buttonGroup drawBtn dynBtns (ButtonGroupConfig pType pTag iVal setV dAtts) = do
+  dAtts' <- mapDyn (Map.insert "type" pType) dAtts
+  (parent, (dynV, internV, child)) <- elDynAttr' pTag dAtts' $ mdo
+    pb <- getPostBuild
+    let externSet = attachWith revLookup (current dynBtns) setV
+        initSet   = attachWith revLookup (current dynBtns) (iVal <$ pb)
+        internSet = leftmost [initSet, clickSelEvts]
+        internVal = attachWith (\m k -> k >>= flip Map.lookup m)
+                              (current dynBtns)
+                              internSet
+    dynK     <- holdDyn Nothing $ leftmost [internSet, externSet]
```

Comment:
I think it's generally better to avoid "Hungarian Notation" style names, i.e. names that mention their types.  I'd probably just call this `k` if i were you.